jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/ContainerBuildPlan.java

```diff
@@ -34,11 +35,13 @@
   public static class Builder {
 
     private String baseImage = "scratch";
-    private String architectureHint = "amd64";
-    private String osHint = "linux";
     private Instant creationTime = Instant.EPOCH;
     private ImageFormat format = ImageFormat.Docker;
 
+    // note that a LinkedHashSet instead of HashSet has been used so as to preserve the platform
+    // order
+    private Set<Platform> platforms = new LinkedHashSet<>();
```

Comment:
```java
    private Set<Platform> platforms = new LinkedHashSet<>(Collections.singleton(new Platform("linux", "amd64")));
```

You can then remove all the "if empty then add amd64/linux as a default" code in multiple places.

Reply:
Good Point